### PR TITLE
[tailscale1.20] cmd/dist: always default to CGO_ENABLED=""

### PR DESCRIPTION
**Pull request by danderson**
Lets us build a statically linked toolchain that has the same cgo default as the standard dynamic toolchain.

Signed-off-by: David Anderson <danderson@tailscale.com>

### CODE REVIEW

File: src/cmd/dist/buildgo.go [hunk 1]
```diff
@@ -114,7 +113,7 @@ func mkzcgo(dir, file string) {
 	fmt.Fprintln(&buf)
 	fmt.Fprintf(&buf, "package build\n")
 	fmt.Fprintln(&buf)
-	fmt.Fprintf(&buf, "const defaultCGO_ENABLED = %q\n", os.Getenv("CGO_ENABLED"))
+	fmt.Fprintf(&buf, "const defaultCGO_ENABLED = %q\n", "1")
```
Comment (bradfitz) [inline src/cmd/dist/buildgo.go]: How about just the empty string? So it'll use its normal heuristics

Comment (danderson) [reply src/cmd/dist/buildgo.go]: Oh, didn't see it had extra behavior for unset. Done.

**Review by bradfitz — APPROVED** [review]

**Comment by bradfitz** [comment]
Commit message is wrong now, though

**Comment by danderson** [comment]
msg fixed.

**Pull request merged**